### PR TITLE
Update Python3 Dependencies and Fix Deployment on Win2016 [2.4.7]

### PR DESCRIPTION
**Pull request by cindydeng1998**
**Task name**: AzureIoTEdgeV2

**Description**: 

- Debug iotedgedev 3.0.0 installation package dependencies 
- Update fixed version of az-cli azure-iot from 0.9 to 0.10.10
- Added fix for win2016 Deployment issue

**Documentation changes required:** (Y/N) N

**Added unit tests:** (Y/N) N
**Attached related issue:** (Y/N) 
- https://github.com/Azure/azure-iot-cli-extension/issues/325 

**Checklist**:
- [x] Task version was bumped - please check [instruction](https://github.com/microsoft/azure-pipelines-tasks/tree/master/docs/taskversionbumping.md) how to do it
- [x] Checked that applied changes work as expected

### CODE REVIEW

**Comment by cindydeng1998** [comment]
@Josh-01 Hi Jyostna, here is the updated task to unblock our customer issue linked above. Can we get your review? Thank you so much! 

@mhshami01 For future reference and review.

**Comment by Josh-01** [comment]
@cindydeng1998
Hi Cindy,
Someone from your team has to approve the code changes before I can approve and merge.

File: Tasks/AzureIoTEdgeV2/task.json
Comment (marianan) [inline Tasks/AzureIoTEdgeV2/task.json:14]: >version [](start = 5, length = 7)

what about the version in the package.json, do we need to increment it as well?

File: Tasks/AzureIoTEdgeV2/deployimage.ts
Comment (marianan) [inline Tasks/AzureIoTEdgeV2/deployimage.ts:50]: >paho-mqtt situation goes away [](start = 98, length = 29)

Could you expand on the root cause and the work around?

File: Tasks/AzureIoTEdgeV2/deployimage.ts
Comment (marianan) [inline Tasks/AzureIoTEdgeV2/deployimage.ts:54]: the setuptools upgrade is extra logic needed by the az cli, i suggest we move it into installAzureCliIotExtension function

File: Tasks/AzureIoTEdgeV2/deployimage.ts
Comment (marianan) [inline Tasks/AzureIoTEdgeV2/deployimage.ts:66]: >fix for windows only [](start = 62, length = 20)

is the setuptools upgrade required because the tools cache has another version? If so, then I expect that the az cli team has a tracking item to update the toolscache on win2016 images, otherwise all scenarios in this environment will require this workaround. Alternatively, can the az cli installation handle this specific requirement as part of its install?

**Review by marianan — CHANGES_REQUESTED** [review]
:clock1:

File: Tasks/AzureIoTEdgeV2/util.ts
Comment (marianan) [inline Tasks/AzureIoTEdgeV2/util.ts:89]: >cmds [](start = 6, length = 4)

Do you know if this change will also address this issue, customers are able to work around using this updated script:
[https://github.com/microsoft/azure-pipelines-tasks/issues/14167](https://github.com/microsoft/azure-pipelines-tasks/issues/14167)
sudo apt install python3-pip
sudo pip3 install --upgrade pip==20.3.4
sudo pip3 install --upgrade cryptography
sudo pip3 install --upgrade iotedgedev
iotedgedev --version

**Comment by marianan** [comment]
>         { path: `pip`, arg: `install ${Constants.iotedgedev}~=${version}`, execOption: Constants.execSyncSilentOption },

should all 'pip' calls be replaced with 'pip3' now?

---
Refers to: Tasks/AzureIoTEdgeV2/util.ts:98 in b08fdfb. [](commit_id = b08fdfb67a178f5ba0ed4021ea01d2175268113f, deletion_comment = False)

File: Tasks/AzureIoTEdgeV2/util.ts
Comment (cindydeng1998) [reply Tasks/AzureIoTEdgeV2/util.ts:89]: I've only tested the script below on 18.04 linux and 16.04 linux, I can give these a try and test out 20.04 today

File: Tasks/AzureIoTEdgeV2/deployimage.ts
Comment (cindydeng1998) [reply Tasks/AzureIoTEdgeV2/deployimage.ts:50]: The root cause is that sometimes the hosted agent does not recognize that the azure-iot extension is installed so the temporary workaround is to verify the version that is currently installed for sanity check. It doesn't actually solve the issue within the task, only works as a devops script.

File: Tasks/AzureIoTEdgeV2/deployimage.ts
Comment (cindydeng1998) [reply Tasks/AzureIoTEdgeV2/deployimage.ts:66]: The setuptools command is because az-cli runs in its own private python environment and thus not relevant to the version installed in the toolscache.

File: Tasks/AzureIoTEdgeV2/task.json
Comment (cindydeng1998) [reply Tasks/AzureIoTEdgeV2/task.json:14]: @Josh-01 do we need to update version numbers in package.json when we do updates to an extension?

**Review by marianan — APPROVED** [review]
:shipit:

File: Tasks/AzureIoTEdgeV2/task.json
Comment (Josh-01) [reply Tasks/AzureIoTEdgeV2/task.json:14]: Just the task.json version update should do.

**Review by Josh-01 — APPROVED** [review]

**Comment by Josh-01** [comment]
Checks not running.

**Pull request closed without merging**
